examples/mlflow-3/evaluate_example.py

```diff
@@ -0,0 +1,26 @@
+from sklearn.datasets import load_iris
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+import mlflow
+from mlflow.models import infer_signature
+
+X, y = load_iris(return_X_y=True, as_frame=True)
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+model = LogisticRegression().fit(X_train, y_train)
+
+predictions = model.predict(X_train)
+signature = infer_signature(X_train, predictions)
+
+with mlflow.start_run() as run:
+    logged_model = mlflow.sklearn.log_model(model, "model", signature=signature)
+    model_uri = f"models:/{logged_model.model_id}"
+    result = mlflow.evaluate(
+        model_uri,
```

Comment:
Can we also handle / test the case where I've loaded a pyfunc model and I pass the pyfunc model instance to `mlflow.evaluate()`? It would be great if the model ID was attached to the pyfunc model instance

Reply:
Will stamp this PR - feel free to add this as a followup ^ :)

Reply:
+1 for a follow-up. I'll merge this once other comments are addressed.